enterprise/cypher/slotted-runtime/src/main/java/org/neo4j/cypher/internal/compatibility/v3_3/runtime/slotted/DefaultComparatorTopTable.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_3.runtime.slotted;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.PriorityQueue;
+
+/**
+ * The default implementation of a Top N table used by the generated code
+ *
+ * It accepts tuples as boxed objects that implements Comparable
+ *
+ * Implements the following interface:
+ * (since the code is generated it does not actually need to declare it with implements)
```

Comment:
This comment seems to have been copied from classes in the compiled runtime. Should be reworded or deleted.